src/algebra/big_operators/finprod.lean

```diff
@@ -566,6 +566,20 @@ end
   ∏ᶠ i ∈ (insert a s), f i = ∏ᶠ i ∈ s, f i :=
 finprod_mem_insert_of_eq_one_if_not_mem (λ _, h)
 
+/-- If the multiplicative support of `f` is finite, then for every `x` in the domain of `f`,
+`f x` divides `finprod f`.  -/
+lemma finprod_mem_dvd {f : α → N} (a : α) (hf : finite (mul_support f)) :
+  (f a) ∣ (finprod f) :=
+begin
+  classical,
+  by_cases ha: a ∈ (mul_support f),
+  { rw [finprod_eq_prod_of_mul_support_to_finset_subset f hf (set.subset.refl _),
+      finset.prod_eq_mul_prod_diff_singleton ((finite.mem_to_finset hf).mpr ha) f],
+    use ∏ (x : α) in hf.to_finset \ {a}, f x },
```

Comment:
I think we should extract a finset proof here, something like:
```lean
lemma finset.dvd_prod_of_mem (f : α → N) (a : α) (s : finset α) (ha : a ∈ s) :
  f a ∣ ∏ i in s, f i :=
begin
  classical,
  rw finset.prod_eq_mul_prod_diff_singleton ha,
  exact dvd_mul_right _ _,
end
```
which makes this
```lean
/-- If the multiplicative support of `f` is finite, then for every `x` in the domain of `f`,
`f x` divides `finprod f`.  -/
lemma dvd_finprod {f : α → N} (a : α) (hf : finite (mul_support f)) :
  (f a) ∣ (finprod f) :=
begin
  by_cases ha : a ∈ mul_support f,
  { rw [finprod_eq_prod_of_mul_support_to_finset_subset f hf (set.subset.refl _)],
    exact finset.dvd_prod_of_mem f _ _ ((finite.mem_to_finset hf).mpr ha) },
  { rw nmem_mul_support.1 ha,
    exact one_dvd (finprod f) }
end
```